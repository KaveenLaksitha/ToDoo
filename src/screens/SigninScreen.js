import { View, StyleSheet, Pressable, Text } from 'react-native'
import React, { useEffect } from 'react'
import { GoogleSignin, statusCodes, GoogleSigninButton } from '@react-native-google-signin/google-signin';

const SigninScreen = ({ navigation }) => {

    useEffect(() => {
        GoogleSignin.configure({
            webClientId: '354041699727-c21srjuqmgs9iqv2t70brfc5fqmpgmpk.apps.googleusercontent.com', // client ID of type WEB for your server (needed to verify user ID and offline access)
            offlineAccess: true, // if you want to access Google API on behalf of the user FROM YOUR SERVER
        });
    }, []);

    signIn = async () => {
        try {
            await GoogleSignin.hasPlayServices();
            const userInfo = await GoogleSignin.signIn();
            console.log("data>>>", userInfo)
        } catch (error) {
            console.log(error)
            if (error.code === statusCodes.SIGN_IN_CANCELLED) {
                // user cancelled the login flow
                console.log("cancelled")
            } else if (error.code === statusCodes.IN_PROGRESS) {
                // operation (e.g. sign in) is in progress already
                console.log("in progress")

            } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
                // play services not available or outdated
                console.log("in progress")
            } else {
                // some other error happened
                console.log("error", error);
            }
        }
    };

    return (
        <View style={styles.container}>
            <Text style={styles.header}>Welcome to ToDoo</Text>
            <GoogleSigninButton
                onPress={() => { signIn() }}
                style={{ width: 192, height: 48 }}
                size={GoogleSigninButton.Size.Wide}
                color={GoogleSigninButton.Color.Dark}
            // style={styles.signinbtn}
            />
            <Pressable onPress={() => { navigation.replace('home') }} style={styles.signinbtn}>
                <Text style={styles.text}>home</Text>
            </Pressable>
        </View>
    )
}


//styles
const styles = StyleSheet.create({
    container: {
        flex: 0.5,
        alignItems: "center",
        justifyContent: "center"
    },
    header: {
        textAlign: "center",
        fontWeight: "bold",
        fontSize: 36,
        color: 'black'
    },
    signinbtn: {
        backgroundColor: 'red',
        padding: 10
    },
    text: {
        textAlign: "center",
        fontWeight: "bold",
        fontSize: 18,
        color: 'white'
    }
})

export default SigninScreen