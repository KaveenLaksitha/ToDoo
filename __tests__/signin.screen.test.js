import React from 'react'
import SigninScreen from '../src/screens/SigninScreen'

import { render, fireEvent } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

describe('Signin Screen', () => {

    it('should render image in the signin screen', () => {
        const { getByTestId } = render(<SigninScreen />)

        const image = getByTestId('image')
        expect(image).toBeTruthy()
    });

    it('should render email input in the signin screen', () => {
        const { getByTestId } = render(<SigninScreen />)

        const input = getByTestId('textInput-email')
        expect(input).toBeTruthy()
    });

    it('should render password input in the signin screen', () => {
        const { getByTestId } = render(<SigninScreen />)

        const input = getByTestId('textInput-password')
        expect(input).toBeTruthy()
    });


    it('should render signin button in the signin screen', () => {
        const page = render(<SigninScreen />)

        const loginBtn = page.getByTestId('loginButton')
        expect(loginBtn).toBeTruthy()

    });

    it('should work signin button in the signin screen correctly', () => {
        const { getByTestId } = render(<SigninScreen />)

        const loginBtn = getByTestId('loginButton')
        fireEvent.press(loginBtn)

    });

    // it('should go to home screen on login button press', () => {
    //     const navigation = { navigate: () => { } }
    //     spyOn(navigation, 'navigate')

    //     const page = render(<SigninScreen navigation={navigation} />)

    //     const loginBtn = page.getByTestId('loginButton')

    //     fireEvent.press(loginBtn)

    //     expect(navigation.navigate).toHaveBeenCalledWith('home');
    // })

    test('renders sign in page correctly ', () => {
        const tree = renderer.create(<SigninScreen />).toJSON()
        expect(tree).toMatchSnapshot()
    })

});
