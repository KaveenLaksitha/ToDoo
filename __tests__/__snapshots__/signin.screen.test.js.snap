// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Screen renders sign in page correctly  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 36,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Welcome to ToDoo
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 15,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 250,
          "resizeMode": "cover",
          "width": 250,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://i.ibb.co/jRhgq3W/undraw-Shared-goals-re-jvqd.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 250,
              "width": 250,
            },
            undefined,
          ]
        }
        testID="image"
      />
    </View>
    <View>
      <TextInput
        onChangeText={[Function]}
        placeholder="email"
        placeholderTextColor="grey"
        style={
          Object {
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "black",
            "fontSize": 16,
            "marginBottom": 20,
            "padding": 10,
            "width": 350,
          }
        }
        testID="textInput-email"
      />
    </View>
    <View>
      <TextInput
        onChangeText={[Function]}
        placeholder="password"
        placeholderTextColor="grey"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "black",
            "fontSize": 16,
            "marginBottom": 20,
            "padding": 10,
            "width": 350,
          }
        }
        testID="textInput-password"
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#52007a",
        "borderRadius": 15,
        "justifyContent": "center",
        "padding": 10,
        "width": "85%",
      }
    }
    testID="loginButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Signin
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "black",
        }
      }
    >
      Don't have a account? 
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "blue",
          }
        }
      >
        Signup
      </Text>
    </View>
  </View>
</View>
`;
